test/EngineTest.cpp

```diff
@@ -192,47 +330,111 @@ TEST(EngineTest, optionalJoinTest) {
   resWidth = vres.numColumns();
   CALL_FIXED_SIZE((std::array{aWidth, bWidth, resWidth}),
                   OptionalJoin::optionalJoin, va, vb, true, false, jcls, &vres);
-
-  ASSERT_EQ(5u, vres.size());
-  ASSERT_EQ(7u, vres.numColumns());
-
-  vector<Id> r{I(1), I(2), I(3), I(4), I(5), I(6), I(4)};
-  for (size_t i = 0; i < 7; i++) {
-    ASSERT_EQ(r[i], vres[0][i]);
-  }
-  r = {I(1), I(2), I(3), I(4), I(5), I(6), I(5)};
-  for (size_t i = 0; i < 7; i++) {
-    ASSERT_EQ(r[i], vres[1][i]);
-  }
-  r = {I(1), I(2), I(3), I(7), I(5), I(6), I(4)};
-  for (size_t i = 0; i < 7; i++) {
-    ASSERT_EQ(r[i], vres(2, i));
-  }
-  r = {I(1), I(2), I(3), I(7), I(5), I(6), I(5)};
-  for (size_t i = 0; i < 7; i++) {
-    ASSERT_EQ(r[i], vres(3, i));
-  }
-  r = {ID_NO_VALUE, I(6), I(7), ID_NO_VALUE, ID_NO_VALUE, ID_NO_VALUE, I(4)};
-  for (size_t i = 0; i < 7; i++) {
-    ASSERT_EQ(r[i], vres(4, i));
-  }
+  
+  // For easier checking.
+  sampleSolution = makeIdTableFromVector(std::vector<std::vector<size_t>>{
+          {1, 2, 3, 4, 5, 6, 4},
+          {1, 2, 3, 4, 5, 6, 5},
+          {1, 2, 3, 7, 5, 6, 4},
+          {1, 2, 3, 7, 5, 6, 5},
+          {0, 6, 7, 0, 0, 0, 4}
+        }
+      );
+  sampleSolution(4, 0) = ID_NO_VALUE;
+  sampleSolution(4, 3) = ID_NO_VALUE;
+  sampleSolution(4, 4) = ID_NO_VALUE;
+  sampleSolution(4, 5) = ID_NO_VALUE;
+
+  ASSERT_EQ(sampleSolution.size(), vres.size());
+  ASSERT_EQ(sampleSolution.numColumns(), vres.numColumns());
+  ASSERT_EQ(sampleSolution, vres);
 }
 
 TEST(EngineTest, distinctTest) {
-  IdTable inp(4, allocator());
-  IdTable res(4, allocator());
+  IdTable inp = makeIdTableFromVector(std::vector<std::vector<size_t>>{{{1, 1, 3, 7}, {6, 1, 3, 6}, {2, 2, 3, 5}, {3, 6, 5, 4}, {1, 6, 5, 1}}});
 
-  inp.push_back({I(1), I(1), I(3), I(7)});
-  inp.push_back({I(6), I(1), I(3), I(6)});
-  inp.push_back({I(2), I(2), I(3), I(5)});
-  inp.push_back({I(3), I(6), I(5), I(4)});
-  inp.push_back({I(1), I(6), I(5), I(1)});
+  IdTable res(4, allocator());
 
   std::vector<size_t> keepIndices = {1, 2};
   CALL_FIXED_SIZE(4, Engine::distinct, inp, keepIndices, &res);
-
-  ASSERT_EQ(3u, res.size());
-  ASSERT_EQ(inp[0], res[0]);
-  ASSERT_EQ(inp[2], res[1]);
-  ASSERT_EQ(inp[3], res[2]);
+  
+  // For easier checking.
+  IdTable sampleSolution = makeIdTableFromVector(std::vector<std::vector<size_t>>{{{1, 1, 3, 7}, {2, 2, 3, 5}, {3, 6, 5, 4}}});
+  ASSERT_EQ(sampleSolution.size(), res.size());
```

Comment:
The `ASSERT_EQ()` of the sizes is not necessary, because that is also checked with the next line.

Reply:
Done, see [97c2f52](https://github.com/ad-freiburg/qlever/pull/821/commits/97c2f522a384423656544e24abf410da607de250).